packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -251,6 +279,9 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
                 })
             ]);
         }
+        this.toDispose.push(this.corePreferences.onPreferenceChanged(() => {
```

Comment:
@kaiyue0329 we can update only when we need to by checking first if the preference we care about has been updated:

```ts
this.toDispose.push(this.corePreferences.onPreferenceChanged(preference => {
  if (preference.preferenceName === 'workbench.tree.renderIndentGuides') {
      this.update();
  }
}));
```